src/utils/region.ts

```diff
@@ -1,9 +1,13 @@
 import { Region } from 'types/general/region';
 
-export const queryBuilderAddRegion = (query: object | undefined, region?: Region) => {
+export const addRegionToQuery = (query: object | undefined, region?: Region) => {
   if (region) {
     return { ...query, region };
   }
 
   return query;
 };
+
+export const getRegionFromString = (region?: any): Region | undefined => {
```

Comment:
I would throw an error instead of undefined.

This function is called only when the user INPUTS a region. if the region does not match any value of the enum, throw error, instead of invoking the server, and getting 400.

Reply:
We talked.